cli/src/config.rs

```diff
@@ -229,20 +231,48 @@ impl UnresolvedConfigEnv {
             config_dir.push("conf.d");
             ConfigPath::new(config_dir)
         });
+
         use ConfigPath::*;
-        if let Some(path @ Existing(_)) = home_config_path {
-            paths.push(path);
-        } else if let (Some(path @ New(_)), None) = (home_config_path, &platform_config_path) {
-            paths.push(path);
+
+        match home_config_path {
+            Some(path @ Existing(_)) => paths.push(path),
+            Some(path @ New(_)) if platform_config_path.is_none() => paths.push(path),
+            _ => (),
+        }
+
+        #[cfg(target_os = "macos")]
+        {
+            let legacy_platform_config_path =
+                self.macos_legacy_config_dir.clone().map(|mut config_dir| {
+                    config_dir.push("jj");
+                    config_dir.push("config.toml");
+                    ConfigPath::new(config_dir)
+                });
+            let legacy_platform_config_dir = self.macos_legacy_config_dir.map(|mut config_dir| {
+                config_dir.push("jj");
+                config_dir.push("conf.d");
+                ConfigPath::new(config_dir)
+            });
+
+            if let Some(path @ Existing(_)) = legacy_platform_config_path {
+                paths.push(path);
+            }
+            if let Some(path @ Existing(_)) = legacy_platform_config_dir {
+                paths.push(path);
+            }
         }
+
         // This should be the default config created if there's
         // no user config and `jj config edit` is executed.
+        // NOTE: this should probably only be pushed if it exists, or no other config
```

Comment:
I made a note here so I could bring it up - it seems unfortunate that if you already have a `~/.jjconfig.toml`, for example, when you use `jj config edit --user`, it will pop a chooser for either `~/.jjconfig.toml` _or_ `~/.config/jj/config.toml`, even if the latter does not exist.

Reply:
Sounds like a bug. Existing paths should be prioritized over new paths.

Reply:
It needs to be pushed if there's no other configs as `jj config edit --user` will try to automatically create it

Reply:
Right, but it shouldn't be pushed if there are other configs.

Reply:
I would agree with that

Reply:
Opened a PR: #6315